source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift

```diff
@@ -0,0 +1,159 @@
+            view.inputString = inputElement.getValue()?.stringValue ?? ""
+            return view
+        }()
+
+        inputField.setNumericFormatter(InputNumberFormatter())
+        inputField.stepperShouldWrapValues(false)
+        inputField.isHidden = !inputElement.getIsVisible()
+        
+        return inputField
+    }
+}
+
+// MARK: - ACRNumericTextField Class
+open class ACRNumericTextField: NSView, NSTextFieldDelegate {
+    override init(frame frameRect: NSRect) {
+        super.init(frame: frameRect)
+        setupViews()
+        setupConstraints()
+        setUpControls()
+        setUpTheme()
+    }
+    
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private lazy var textField: NSTextField = {
+        let view = NSTextField()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.delegate = self
+        view.isEditable = true
+        return view
+    }()
+
+    private lazy var stepper: NSStepper = {
+        let view = NSStepper()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    @objc private func handleStepperAction(_ sender: NSStepper) {
+        textField.stringValue = "\(sender.integerValue)"
+    }
+    
+    public func controlTextDidChange(_ obj: Notification) {
+        guard let textView = obj.object as? NSTextField else { return }
+        // updating the stepper value when text field value change
+        stepper.doubleValue = Double(textView.stringValue) ?? stepper.doubleValue
+    }
+    
+    func setupViews() {
+        addSubview(textField)
+        addSubview(stepper)
+    }
+    
+    func setupConstraints() {
+        textField.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        textField.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        textField.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+        stepper.leadingAnchor.constraint(equalTo: textField.trailingAnchor).isActive = true
+        stepper.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        stepper.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+        stepper.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+    }
+    
+    func setUpControls() {
+        stepper.target = self
+        stepper.action = #selector(handleStepperAction(_:))
+    }
+    
+    func setUpTheme() {
+        textField.wantsLayer = true
+        textField.layer?.backgroundColor = .clear
+        textField.appearance = NSAppearance(named: .aqua)
+        stepper.wantsLayer = true
+        stepper.layer?.backgroundColor = .clear
+        stepper.appearance = NSAppearance(named: .aqua)
+    }
+}
+
+// MARK: - EXTENSION
+extension ACRNumericTextField {
+    static let MAXVAL = Double.greatestFiniteMagnitude
+    static let MINVAL = -MAXVAL
```

Comment:
Hey Manu
Is this the way to get the minimum value?
I'm not really sure, plus this is a trial comment

Reply:
Yeah. One of the ways. 
Idea is to basically put a limit on the stepper values with a range allowed by Double from -ve to +ve side if not specified explicitly.
The stepper value can be in decimal and negative too as per the current implementation. 
Let me know if this can be handled differently.